src/NHibernate/Linq/Visitors/QueryModelVisitor.cs

```diff
@@ -121,6 +122,9 @@ public class QueryModelVisitor : NhQueryModelVisitorBase, INhQueryModelVisitor,
 
 		public ResultOperatorRewriterResult RewrittenOperatorResult { get; private set; }
 
+		internal Dictionary<NhJoinClause, FetchRequestBase> RelatedJoinFetchRequests { get; } =
+			new Dictionary<NhJoinClause, FetchRequestBase>();
```

Comment:
Same question.. Shouldn't it be restricted to `FetchOneRequest`?

Reply:
It should, restricted.